### PR TITLE
improve intro for 'sources' in style spec

### PR DESCRIPTION
**Pull request by domlet**
improves clarity and adds a header.

Tagging @ryanhamley to verify (or correct for) accuracy:

Screen Shot 2021-01-26 at 1 35 07 PM

### CODE REVIEW

File: docs/pages/style-spec/sources.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ A map or layer **source** states which data the map should display. Specify the
     return [prev, ', ', curr]
 })}}. 
 
-A [source]() provides map data which Mapbox GL JS can use with a [style]() document to help render a client render a visual representation of that data. This delegation makes it possible to style the same source in different ways, as you might do to differentiate the appearances of different types of roads in a highways layer.
+A [source](https://docs.mapbox.com/help/glossary/source) provides map data which Mapbox GL JS can use with a [style](https://docs.mapbox.com/help/glossary/style) document to help render a client render a visual representation of that data. This delegation makes it possible to style the same source in different ways, as you might do to differentiate the appearances of different types of roads in a highways layer.
```
Comment (ryanhamley) [inline docs/pages/style-spec/sources.md]: There's a typo in this sentence: `to help render a client render a visual representation`. I think it reads best as `to help render a visual representation`.

**Review by ryanhamley — APPROVED** [review]
Looks good! We should take this opportunity to fix the typo though.

File: docs/pages/style-spec/sources.md (hunk 1 above)
Comment (domlet) [inline docs/pages/style-spec/sources.md]:
```suggestion
A [source](https://docs.mapbox.com/help/glossary/source) provides map data which Mapbox GL JS can use with a [style](https://docs.mapbox.com/help/glossary/style) document to help render a visual representation of that data. This delegation makes it possible to style the same source in different ways, as you might do to differentiate the appearances of different types of roads in a highways layer.
```

Comment (domlet) [reply docs/pages/style-spec/sources.md]: Wonderful; this is now implemented in b6627cc – thanks Ryan for your careful reviews!

**Comment by domlet** [comment]
Tagging @HeyStenson for docs review – and wondering if we should be referring to "style specifications" or "styling instructions" instead of "use with a style document" or "Specify a style" (in the Note) because those latter two phrases are biased towards a Mapbox-hosted style rather than style instructions provided via GL JS. Thank you both for your reviews!

File: docs/pages/style-spec/sources.md [hunk 2]
```diff
@@ -12,17 +12,24 @@ prependJs:
...
 <!--copyeditor disable macos-->
-Sources state which data the map should display. Specify the type of source with the `"type"` property, which must be one of {{sourceTypes.map((t, i) => {
+A map or layer **source** states which data the map should display. Specify the type of source with the `"type"` property, which must be one of {{sourceTypes.map((t, i) => {
     return <var key={i}>{t}</var>;
 }).reduce((prev, curr) => {
     return [prev, ', ', curr]
-})}}. Adding a source isn't enough to make data appear on the map because sources don't contain styling details like color or width. Layers refer to a source and give it a visual representation. This makes it possible to style the same source in different ways, like differentiating between types of roads in a highways layer.
+})}}. 
+
+A [source](https://docs.mapbox.com/help/glossary/source) provides map data which Mapbox GL JS can use with a [style](https://docs.mapbox.com/help/glossary/style) document to help render a visual representation of that data. This delegation makes it possible to style the same source in different ways, as you might do to differentiate the appearances of different types of roads in a highways layer.
```
Comment (HeyStenson) [inline docs/pages/style-spec/sources.md:28]:
```suggestion
A [source](https://docs.mapbox.com/help/glossary/source) provides map data that Mapbox GL JS can use with a [style](https://docs.mapbox.com/help/glossary/style) document to render a visual representation of that data. This delegation makes it possible to style the same source in different ways, as you might do to differentiate the appearances of different types of roads in a highways layer.
```

**Review by HeyStenson — APPROVED** [review]
> wondering if we should be referring to "style specifications" or "styling instructions" instead of "use with a style document" or "Specify a style" (in the Note) because those latter two phrases are biased towards a Mapbox-hosted style rather than style instructions provided via GL JS. 

Hmm! I wouldn't necessarily agree that "style document" or "specify a style" imply a Mapbox-hosted style, though maybe I've just looked at that particular way of wording things too long 🙈  But I do agree that it would be a useful exercise to agree on a standard way to phrase this and then go on a docs-wide quest to make updates! It'd be interesting to get feedback from the whole team about this issue.

File: docs/pages/style-spec/sources.md [hunk 3]
```diff
@@ -12,17 +12,24 @@ prependJs:
...
     return <var key={i}>{t}</var>;
 }).reduce((prev, curr) => {
     return [prev, ', ', curr]
-})}}. Adding a source isn't enough to make data appear on the map because sources don't contain styling details like color or width. Layers refer to a source and give it a visual representation. This makes it possible to style the same source in different ways, like differentiating between types of roads in a highways layer.
+})}}. 
+
+A [source](https://docs.mapbox.com/help/glossary/source) provides map data that Mapbox GL JS can use with a [style](https://docs.mapbox.com/help/glossary/style) document to render a visual representation of that data. This delegation makes it possible to style the same source in different ways, as you might do to differentiate the appearances of different types of roads in a highways layer.
+
+{{ <Note title="Specify a style"><p>Adding a source to a map or layer isn't enough to make data appear on the map. You must also specify a style to provide details like color or width for a feature.</p></Note> }}
```
Comment (domlet) [inline docs/pages/style-spec/sources.md:30]:
```suggestion
{{ <Note title="Specify a style"><p>Adding a source to a map or layer isn't enough to make data appear on the map. You must also specify a style to provide details like color or width for a feature.</p></Note> }}
```
```suggestion
{{ <Note title="Specify a style"><p>Adding a source to a map or layer isn't enough to make data appear on the map. You must also specify a style to provide properties like color or width for each feature.</p></Note> }}
```

**Pull request merged**